tests/e2e/e2e_setup_test.go

```diff
@@ -210,15 +247,22 @@ func (s *IntegrationTestSuite) SetupSuite() {
 		}
 	}
 
-	if !s.skipUpgrade {
+	switch {
+	case !s.skipUpgrade && !s.isFork:
 		s.createPreUpgradeState()
 		s.upgrade()
 		s.runPostUpgradeTests()
+	case !s.skipUpgrade && s.isFork:
+		s.createPreUpgradeState()
+		s.upgradeFork()
+		s.runPostUpgradeTests()
+	case s.skipUpgrade:
+		s.runPostUpgradeTests()
```

Comment:
`s.runPostUpgradeTests()` should be in its own fixture `func TestPostUpgrade(t *testing.T)` in `e2e_test.go`. We can do this in a future PR